src/main/kotlin/com/openlattice/data/DataGraphService.kt

```diff
@@ -169,71 +176,197 @@ open class DataGraphService(
-        logger.info("Finished deleting {} edges.", edgesDeletedCount)
-        return eds.deleteEntitySetData(entitySetId, authorizedPropertyTypes)
+        // delete associations
+        deleteAssociations(entitySetId, Optional.empty())
+
+        // delete edges
+        val verticesCount = graphService.deleteVerticesInEntitySet(entitySetId)
+
+        // delete entities
+        val entityCount = eds.deleteEntitySetData(entitySetId, authorizedPropertyTypes)
+
+        logger.info("Deleted {} entities and {} vertices.", entityCount, verticesCount)
+        return entityCount
     }
 
     override fun deleteEntities(
             entitySetId: UUID,
             entityKeyIds: Set<UUID>,
             authorizedPropertyTypes: Map<UUID, PropertyType>
     ): Int {
-        val verticesCount = graphService.deleteVertices(entitySetId, entityKeyIds)
+        // delete associations
+        deleteAssociations(entitySetId, Optional.of(entityKeyIds))
+
+        return deleteEntityDataAnVertices(entitySetId, entityKeyIds, authorizedPropertyTypes)
+    }
+
+    private fun deleteEntityDataAnVertices(
+            entitySetId: UUID,
+            entityKeyIds: Set<UUID>,
+            authorizedPropertyTypes: Map<UUID, PropertyType>): Int {
+        // delete edges
         val entityCount = eds.deleteEntities(entitySetId, entityKeyIds, authorizedPropertyTypes)
+
+        // delete entities
+        val verticesCount = graphService.deleteVertices(entitySetId, entityKeyIds)
+
         logger.info("Deleted {} entities and {} vertices.", entityCount, verticesCount)
+
         return entityCount
     }
 
-    @Timed
-    override fun deleteAssociation(keys: Set<EdgeKey>, authorizedPropertyTypes: Map<UUID, PropertyType>): Int {
-        val entitySetsToEntityKeyIds = HashMultimap.create<UUID, UUID>()
+    private fun deleteAssociations(entitySetId: UUID, entityKeyIds: Optional<Set<UUID>>) {
+        // collect association entity key ids
+        val associationsEntityKeyIds = collectAssociationsEntityKeyIds(entitySetId, entityKeyIds)
+
+        // access checks
+        val authorizedPropertyTypesOfAssociations = HashMap<UUID, Map<UUID, PropertyType>>()
+        associationsEntityKeyIds.forEach {
+            val authorizedPropertyTypesOfAssociation = getAuthorizedPropertyTypesForDelete(it.key, Optional.empty())
+            authorizedPropertyTypesOfAssociations[it.key] = authorizedPropertyTypesOfAssociation
+        }
+
+        // delete associations of entity set
+        val associationDeleteCount = associationsEntityKeyIds.map {
+            deleteEntityDataAnVertices(
+                    it.key,
+                    it.value,
+                    authorizedPropertyTypesOfAssociations[it.key]!!)
+        }.sum()
+
+        logger.info("Deleted {} associations when deleting entities from entity set {}",
+                associationDeleteCount, entitySetId)
+    }
+
+    private fun collectAssociationsEntityKeyIds(
```

Comment:
This is dangerous from a memory perspective. Highly linked objects will OOM the server. It possible re-work to use a Sequence and PostgresIterables.